### PR TITLE
Patches a few bugs in release notes script

### PR DESCRIPTION
**Pull request by urimandujano**
The output format for `gh auth status --show-token` seems to have changed so I updated the regex to pull the token automatically. This at least saved me some copy-pasting everytime I wanted to generate the notes.

### Checklist

- [ ] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [ ] This pull request includes tests or only affects documentation.
- [ ] This pull request includes a label categorizing the change e.g. `fix`, `feature`, `enhancement`, `docs`.
  

For documentation changes:

- [ ] This pull request includes redirect settings in `netlify.toml` for files that are removed or renamed

### CODE REVIEW

File: scripts/generate-release-notes.py [hunk 1]
```diff
@@ -62,7 +62,7 @@ def generate_release_notes(
     )
     if not response.status_code == 200:
         print(
-            "Received status code {response.status_code} from GitHub API:",
+            f"Received status code {response.status_code} from GitHub API:",
```
Comment (zzstoatzz) [inline scripts/generate-release-notes.py:65]: F in the chat

File: scripts/generate-release-notes.py [hunk 2]
```diff
@@ -33,7 +33,7 @@
 REPO_NAME = "prefect"
 DEFAULT_TAG = "preview"
 
-TOKEN_REGEX = re.compile(r"Token:\s(.*)")
+TOKEN_REGEX = re.compile(r"Token:\s(.*)[\w\W]")
```
[the remark this answers is not in the document]
Comment (serinamarie) [reply scripts/generate-release-notes.py:36]: @urimandujano Won't the current regex already match the "Token:" part, then the space, and capture everything that follows (the token) until the end of the line, though?

[the remark this answers is not in the document]
Comment (urimandujano) [reply scripts/generate-release-notes.py:36]: Good catch, I shouldn't have committed this. It was part of my trying-to-figure-out-why the token wasn't being matched. Update incoming!

**Review by serinamarie — APPROVED** [review]

**Review by zzstoatzz — APPROVED** [review]
lgtm!

**Pull request merged**